### ISSUE TITLE
Gettext translations not working on debian with Phalcon 2.0.x

### ISSUE THREAD

**Issue by berliner** [phalcon/cphalcon#10212]
Today I have run into a problem when I deployed a phalcon project to a Debian 4.6.3-14 server.
The translation have been setup following the guide here: https://github.com/phalcon/incubator/tree/master/Library/Phalcon/Translate/Adapter

Translations using the Gettext adapter where working without problems on OS X 10.9.5 using this code to initialize the service:

    $translation_config = array(
        'locale' => $config->application->locale,    // e.g. de_DE.UTF-8
        'file' => 'translations',
        'defaultDomain' => 'translations',
        'directory' => $config->framework->translationsDir,
    );
    new TranslateAdapter($translation_config);

This failed silently, meaning that the translations simply didn't work without any errors or warnings.

I had to add the category parameter to make it work:

    $translation_config = array(
        'locale' => $config->application->locale,    // e.g. de_DE.UTF-8
        'file' => 'translations',
        'defaultDomain' => 'translations',
        'directory' => $config->framework->translationsDir,
        'category' => LC_MESSAGES,
    );
    new TranslateAdapter($translation_config);

Not sure if this is a code issue or a documentation issue, or just my lack of previous experience with phalcon and/or gettext.

**Comment by KorsaR-ZN** [comment]
@berliner Please show output commands `locale`

**Comment by berliner** [comment]
locale
    LANG=de_DE.UTF-8
    LANGUAGE=
    LC_CTYPE="de_DE.UTF-8"
    LC_NUMERIC="de_DE.UTF-8"
    LC_TIME="de_DE.UTF-8"
    LC_COLLATE="de_DE.UTF-8"
    LC_MONETARY="de_DE.UTF-8"
    LC_MESSAGES="de_DE.UTF-8"
    LC_PAPER="de_DE.UTF-8"
    LC_NAME="de_DE.UTF-8"
    LC_ADDRESS="de_DE.UTF-8"
    LC_TELEPHONE="de_DE.UTF-8"
    LC_MEASUREMENT="de_DE.UTF-8"
    LC_IDENTIFICATION="de_DE.UTF-8"
    LC_ALL=

    locale -a
    C
    C.UTF-8
    de_DE.utf8
    en_US.utf8
    POSIX

The phalcon version has been built from the 2.0.x branch in git and reports itself as 2.0.1

**Comment by gunesahmet** [comment]
you must run this command on console dpkg-reconfigure locales and choose your languages.

**Comment by KorsaR-ZN** [comment]
@berliner Change value `$config->application->locale` to `de_DE.utf8`

**Comment by berliner** [comment]
I had tried this before, but without the additional `category ` parameter I didn't get any translations working, only the original string. Actually, in my setup, it doesn't seem to make any difference if the locale is set to de_DE.utf8 or to de_DE.UTF-8.

**Comment by berliner** [comment]
@gunesahmet: I have tried the `dpkg-reconfigure locales` and it didn't change anything.

**Comment by gunesahmet** [comment]
public function getText($language)
    {
        if ($language == null) {
            $language = 'en_US';
        }
        $config = array(
            'gettext_locale_dir' => ROOTPATH .'/languages',
            'gettext_text_domain' => 'messages',
            'gettext_catalog_codeset' => 'UTF-8',
            'gettext_locale' => $language . '.UTF-8'
        );
        bind_textdomain_codeset(
            $config['gettext_text_domain'],
            $config['gettext_catalog_codeset']
        );
        bindtextdomain(
            $config['gettext_text_domain'],
            $config['gettext_locale_dir']
        );
        textdomain(
            $config['gettext_text_domain']
        );

        if(PANEL_OS_TYPE == "linux") {
            setlocale(
                LC_MESSAGES,
                $config['gettext_locale']
            );
        }
        else {
            setlocale(
                LC_ALL,
                $config['gettext_locale']
            );
        }
        putenv('LANGUAGE=' . $config['gettext_locale']);
    }

use this function

**Comment by gunesahmet** [comment]
and dont forget this code only work on linux OS if you use windows change 
  setlocale(
            LC_MESSAGES,
            $config['gettext_locale']
        );
with 
    setlocale(
                LC_ALL,
                $config['gettext_locale']
            );

**Comment by berliner** [comment]
I realize that I haven't been precise enought in my original post, so just to clarify: I had translations working in OS X but couldn't get it to work on the debian server. This issue is not about finding a fix for my server, because I have it up and running now by adding the `category` parameter as shown in the second example of my original post.

So thanks for the proposed help. I just wanted to have this documented in case someone else runs into the same problem and/or if this could have been is a code or documentation issue.

So to summarize, the following config works, at least for me, on both OS X and debian:
    $translation_config = array(
        'locale' => $config->application->locale,    // e.g. de_DE.UTF-8
        'file' => 'translations',
        'defaultDomain' => 'translations',
        'directory' => $config->framework->translationsDir,
        'category' => LC_MESSAGES,    // <- without this it didn't work in debian
    );
    new TranslateAdapter($translation_config);

**Comment by sergeyklay** [comment]
Fixed in the `3.0.x` branch.
